Feature #2454 doc overview conv

lisagoodrich opened the pull request:
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
1. Updated Sphinx online editor information: https://metplus.readthedocs.io/en/feature_2454_doc_overview_conv/Contributors_Guide/documentation.html#testing-rst-formatting-in-an-online-editor.
2. Update tables information Numbers 3 thru 5
3. csv files to automatically create tables. https://metplus.readthedocs.io/en/feature_2454_doc_overview_conv/Contributors_Guide/documentation.html#using-csv-files-to-create-tables 
4. Grid tables with line breaks.  https://metplus.readthedocs.io/en/feature_2454_doc_overview_conv/Contributors_Guide/documentation.html#grid-tables 
5. How to create a line break in a list table.  https://metplus.readthedocs.io/en/feature_2454_doc_overview_conv/Contributors_Guide/documentation.html#line-breaks-in-list-tables
6. How to have code within a paragraph. https://metplus.readthedocs.io/en/feature_2454_doc_overview_conv/Contributors_Guide/documentation.html#code-in-a-paragraph
7. Added a section for Duplicate Links to Troubleshooting https://metplus.readthedocs.io/en/feature_2454_doc_overview_conv/Contributors_Guide/documentation.html#warning-duplicate-explicit-target-name 
   
- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
1.  Please review the above for grammar and clarity.
2.  Number 4 Grid tables, please review the links, especially for the raw RST.  I've linked to "main_v12.1" which doesn't exist yet.  Should we link to the branch where this table does exist?  I wasn't sure.  Here is the table branch location: https://met.readthedocs.io/en/feature_2588_install_rewrite/Users_Guide/installation.html#external-library-handling-in-compile-met-all-sh
3.  for number 6 code in a paragraph, I don’t have any sphinx documentation to back it up.  I looked but couldn’t find anything.  Do you have any additions?
 
Thanks for already reviewing the section locations for the table material.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[Yes]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **1/26/24**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [x] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [x] Close the linked issue and delete your feature or bugfix branch from GitHub.

lisagoodrich committed:
adding documentation in different sections

lisagoodrich committed:
adding grid table section

lisagoodrich committed:
fixing links

lisagoodrich committed:
grammar updates

jprestop committed:
Per #2454, updated sections and wording.

jprestop committed:
Per #2454, added a period to the end of a sentence.

jprestop committed:
Per #2454, fixing formatting

jprestop committed:
Per #2454, updating wording

lisagoodrich committed:
adding a section for line breaks in a table

lisagoodrich committed:
adding :code: information

lisagoodrich committed:
trying to fix warning

lisagoodrich committed:
take 2

lisagoodrich committed:
take 3 or 4

lisagoodrich committed:
maybe fixed

lisagoodrich committed:
updating link

lisagoodrich committed:
fixing web link again

lisagoodrich committed:
web link saga continues

jprestop commented:
@lisagoodrich I changed the project from "Coordinated METplus-5.0 Support" to "METplus-Wrappers-6.0.0 Development" and assigned this to "Beta 3".  :)

jprestop committed:
Changed "ReadTheDocs" to "Read the Docs"

jprestop committed:
Updated "main_v" references to be "main_v12.0"

jprestop committed:
Removed references to main_v*, replacing with raw RST 

It is not maintainable to have links to branches which will become old.  Since we can avoid it by adding the raw RST in the documentation, I have removed all references to main_v* in favor of placing the raw RST in the documentation.

jprestop committed:
Modified the "Code in a Paragraph" section

jprestop committed:
Reworded for consistency within the document

jprestop reviewed (approved):
@lisagoodrich Thank you for all of your work on this task.  I reviewed the things that you asked me to review and made five commits, which you can see at the bottom of [this page](https://github.com/dtcenter/METplus/pull/2471/commits).  To see the individual changes click on the hash value link.  For example, for "Reworded for consistency within the document" the hash value link has the text "19b9303".  Please let me know if you have any questions.  Please go ahead and Squash and Merge when you're ready.

jprestop committed:
Added back the link for Sphinx code blocks

lisagoodrich commented:
@jprestop thanks for all your help and guidance on this, especially the links to the different versions.
